### DISCUSSION TITLE
Best way to use an environment variable in app.html?

### DISCUSSION THREAD

**Discussion by DerekAlanGilbert** [nuxt/nuxt#9526] [Questions (Nuxt 2)] [answered]
Before we upgraded > 2.13 we were using process.env.HEAP_ID in a script tag in app.html. That no longer seems to work past nuxt 2.13. Whats the best way to do this going forward?

**Comment by DerekAlanGilbert** [comment, marked as answer]
As of right now our solution was to use the [webpack lodash syntax](https://cli.vuejs.org/guide/html-and-static-assets.html#interpolation) referenced in the vue cli docs. (even though this isn't a vue cli app I figured we are still using the htmlWebpackPlugin)

So we are doing this and it seems to be working. But, i'd like to upgrade to vite when vue 3 is out and i'm not sure that will be supported. So my question remains in nuxt > 2.13 how do you use env variables in app.html?

```
<script>
somescript.load('<%= process.env.SCRIPT_KEY %>')
</script>
```

**Reply by kissu** [reply]
Do you specifically need to use it in `app.html`?

**Reply by DerekAlanGilbert** [reply]
As of right now I think so. This is the script heap provides us. Not sure where else I can put this thing haha 😆. Thanks @kissu for commenting.
```
	<script type="text/javascript">
			;(window.heap = window.heap || []),
				(heap.load = function(e, t) {
					;(window.heap.appid = e), (window.heap.config = t = t || {})
					var r = document.createElement('script')
					;(r.type = 'text/javascript'), (r.async = !0), (r.src = 'https://cdn.heapanalytics.com/js/heap-' + e + '.js')
					var a = document.getElementsByTagName('script')[0]
					a.parentNode.insertBefore(r, a)
					for (
						var n = function(e) {
								return function() {
									heap.push([e].concat(Array.prototype.slice.call(arguments, 0)))
								}
							},
							p = [
								'addEventProperties',
								'addUserProperties',
								'clearEventProperties',
								'identify',
								'resetIdentity',
								'removeEventProperty',
								'setEventProperties',
								'track',
								'unsetEventProperty'
							],
							o = 0;
						o < p.length;
						o++
					)
						heap[p[o]] = n(p[o])
				})
			heap.load('<%= process.env.HEAP_ID %>')
		</script>
```

**Reply by TheAlexLichter** [reply]
I'd suggest to add this script via a Nuxt plugin. Adding it that way ensures you can inject env variables (and way more) as you like 😋
